pandas/core/generic.py

```diff
@@ -4141,7 +4141,26 @@ class   animal   locomotion
     def __getitem__(self, item):
         raise AbstractMethodError(self)
 
-    def _slice(self, slobj: slice, axis: Axis = 0) -> Self:
+    @final
+    def _getitem_slice(self, key: slice) -> Self:
+        """
+        __getitem__ for the case where the key is a slice object.
+        """
+        # _convert_slice_indexer to determine if this slice is positional
+        #  or label based, and if the latter, convert to positional
+        slobj = self.index._convert_slice_indexer(key, kind="getitem")
```

Comment:
I wonder if the fact that this can return an `ndarray` is a bug, and if we fix that, `slobj` would always be a slice (taken from the test `test_loc_getitem_partial_slice_non_monotonicity`):

```ipython
Ib [1]: import pandas as pd
In [2]: obj = pd.Series(
   ...:             [1] * 5,
   ...:             index=pd.DatetimeIndex(
   ...:                 [
   ...:                     pd.Timestamp("2019-12-30"),
   ...:                     pd.Timestamp("2020-01-01"),
   ...:                     pd.Timestamp("2019-12-25"),
   ...:                     pd.Timestamp("2020-01-02 23:59:59.999999999"),
   ...:                     pd.Timestamp("2019-12-19"),
   ...:                 ],
   ...:                 tz=None,
   ...:             ),
   ...:         )
In [2]: obj["2020-01-01":]
2020-01-01 00:00:00.000000000    1
2020-01-02 23:59:59.999999999    1
dtype: int64
```

I.e. it looks like the above slice does `obj[obj.index >= "2020-01-01"]`. I'd expected the result from `obj["2020-01-01":]` to be:

```
2020-01-01 00:00:00.000000000    1
2019-12-25 00:00:00.000000000    1
2020-01-02 23:59:59.999999999    1
2019-12-19 00:00:00.000000000    1
dtype: int64
```

Reply:
The current behavior is definitely intended.

Reply:
Ok, I don’t use DatetimeIndex so much.